File: scope_shared/scope/enums.py
from enum import Enum


class ActivitySuccessType(Enum):
    Yes = "Yes"
    SomethingElse = "SomethingElse"
    No = "No"


class ClinicCode(Enum):
    Breast = "Breast"
    Endocrine = "Endocrine"
    GI = "GI"
    GIPancreatic = "GI – Pancreatic"
    GU = "GU"
    GYN = "GYN"
    HEME = "HEME"
    HEMESickleCell = "HEME – Sickle Cell"
    HNL = "HNL"
    Immunotherapy = "Immunotherapy"
    MelanomaRenal = "Melanoma/Renal"
    Neuro = "Neuro"
    Sarcoma = "Sarcoma"
    TransplantAuto = "Transplant – Auto"
    TransplantAllo = "Transplant – Allo"
    TransplantCART = "Transplant – CAR-T"
    TransplantLTFU = "Transplant – LTFU"
    TransplantTTC = "Transplant – TTC"
    Other = "Other"


class PatientSex(Enum):
    Male = "Male"
    Female = "Female"
    Intersex = "Intersex"


class PatientGender(Enum):
    Male = "Male"
    Female = "Female"
    Transgender = "Transgender"
    NonBinary = "Non-binary/Non-conforming"
    Other = "Other"


class PatientRace(Enum):
    Native = "American Indian or Alaska Native"
    Asian = "Asian or Asian American"
    Black = "Black or African American"
    Pacific = "Native Hawaiian or Other Pacific Islander"
    White = "White"
    Unknown = "Unknown"


class PatientEthnicity(Enum):
    Hispanic = "Hispanic or Latino/Latina/Latinx"
    NotHispanic = "Not Hispanic or Latino/Latina/Latinx"
    Unknown = "Unknown"


class PatientPronoun(Enum):
    HeHim = "He/Him"
    SheHer = "She/Her"
    TheyThem = "They/Them"


class DepressionTreatmentStatus(Enum):
    CoCM = "CoCM"
    CoCMRelapse = "CoCM RP"
    Discharged = "D/C"
    Other = "Other"
    End = "End"


class FollowupSchedule(Enum):
    OneWeek = "1-week follow-up"
    TwoWeeks = "2-week follow-up"
    ThreeWeeks = "3-week follow-up"
    FourWeeks = "4-week follow-up"
    SixWeeks = "6-week follow-up"
    EightWeeks = "8-week follow-up"


class DiscussionFlag(Enum):
    SafetyRisk = "Flag as safety risk"
    Discussion = "Flag for discussion"


class CancerTreatmentRegimen(Enum):
    Surgery = "Surgery"
    Chemotherapy = "Chemotherapy"
    Radiation = "Radiation"
    StemCellTransplant = "Stem Cell Transplant"
    Immunotherapy = "Immunotherapy"
    CART = "CAR-T"
    Endocrine = "Endocrine"
    Surveillance = "Surveillance"
    Other = "Other"


class ProviderRole(Enum):
    SocialWorker = "socialWorker"
    Psychiatrist = "psychiatrist"
    StudyStaff = "studyStaff"


class Referral(Enum):
    Psychiatry = "Psychiatry"
    Psychology = "Psychology"
    PatienttNavigation = "Patient Navigation"
    IntegrativeMedicine = "Integrative Medicine"
    SpiritualCare = "Spiritual Care"
    PalliativeCare = "Palliative Care"
    Other = "Other"


class ReferralStatus(Enum):
    NotReferral = "Not Referred"
    Pending = "Pending"
    Active = "Active"
    Completed = "Completed"


class SessionType(Enum):
    InPerson = "In person"
    Telehealth = "Telehealth"
    Phone = "Phone"
    Group = "Group"
    Home = "Home"


class TreatmentPlan(Enum):
    MaintainPlan = "Maintain current treatment"
    AdjustPlan = "Adjust treatment plan"
    MonitorOnly = "Monitor only"
    NoFollowup = "No further follow-up"
    ReferToCommunity = "Refer to community"


class TreatmentChange(Enum):
    NoChange = "None"
    Medication = "Medication"
    Counseling = "Counseling"


class BehavioralActivationChecklist(Enum):
    ReviewModel = "Review of the BA model"
    ValuesGoals = "Values and goals assessment"
    ActivityScheduling = "Activity scheduling"
    MoodActivityMonitoring = "Mood and activity monitoring"
    Relaxation = "Relaxation"
    PositiveReinforcement = "Positive reinforcement"
    ManagingAvoidance = "Managing avoidance behaviors"
    ProblemSolving = "Problem-solving"


class BehavioralStrategyChecklist(Enum):
    BehavioralActivation = "Behavioral Activation"
    MotivationalInterviewing = "Motivational Interviewing"
    ProblemSolvingTherapy = "Problem Solving Therapy"
    CognitiveTherapy = "Cognitive Therapy"
    MindfulnessStrategies = "Mindfulness Strategies"
    SupportiveTherapy = "Supportive Therapy"
    Other = "Other"


class AssessmentType(Enum):
    PHQ9 = "phq-9"
    GAD7 = "gad-7"
    MoodLogging = "mood"
    Medication = "medication"


class ScheduledItemFrequency(Enum):
    Daily = "Daily"
    Weekly = "Once a week"
    Biweekly = "Every 2 weeks"
    Monthly = "Every 4 weeks"


class DayOfWeek(Enum):
    Monday = "Monday"
    Tuesday = "Tuesday"
    Wednesday = "Wednesday"
    Thursday = "Thursday"
    Friday = "Friday"
    Saturday = "Saturday"
    Sunday = "Sunday"


class DueType(Enum):
    Exact = "Exact"
    ChunkOfDay = "ChunkOfDay"
    Day = "Day"
    Week = "Week"


class Site(Enum):
    SCCASLU = "SCCA – SLU"
    SCCAUWNW = "SCCA – UW-NW"
    SCCAPEN = "SCCA – PEN"
    MultiCareTacoma = "MultiCare – Tacoma"
    MultiCareGigHarbor = "MultiCare – Gig Harbor"

File: web_shared/enums.ts
export type OtherSpecify = 'Other';

export const patientSexValues = ['Male', 'Female', 'Intersex'] as const;
export type PatientSex = typeof patientSexValues[number];

export const patientGenderValues = ['Male', 'Female', 'Transgender', 'Non-binary/Non-conforming', 'Other'] as const;
export type PatientGender = typeof patientGenderValues[number];

export const patientPronounValues = ['He/Him', 'She/Her', 'They/Them'] as const;
export type PatientPronoun = typeof patientPronounValues[number];

export const patientRaceValues = [
    'American Indian or Alaska Native',
    'Asian or Asian American',
    'Black or African American',
    'Native Hawaiian or Other Pacific Islander',
    'White',
    'Unknown',
] as const;
export type PatientRace = typeof patientRaceValues[number];
export type PatientRaceFlags = { [item in PatientRace]: boolean };

export const patientEthnicityValues = [
    'Hispanic or Latino/Latina/Latinx',
    'Not Hispanic or Latino/Latina/Latinx',
    'Unknown',
] as const;
export type PatientEthnicity = typeof patientEthnicityValues[number];

export const clinicCodeValues = [
    'Breast',
    'Endocrine',
    'GI',
    'GI – Pancreatic',
    'GU',
    'GYN',
    'HEME',
    'HEME – Sickle Cell',
    'HNL',
    'Immunotherapy',
    'Melanoma/Renal',
    'Neuro',
    'Sarcoma',
    'Transplant – Auto',
    'Transplant – Allo',
    'Transplant – CAR-T',
    'Transplant – LTFU',
    'Transplant – TTC',
    'Other',
] as const;
export type ClinicCode = typeof clinicCodeValues[number];

export type AllClinicCode = 'All Clinics';

export const depressionTreatmentStatusValues = ['CoCM', 'CoCM RP', 'D/C', 'Other', 'End'] as const;
export type DepressionTreatmentStatus = typeof depressionTreatmentStatusValues[number];

export const followupScheduleValues = ['1-week follow-up', '2-week follow-up', "3-week follow-up", "4-week follow-up", "6-week follow-up", "8-week follow-up"] as const;
export type FollowupSchedule = typeof followupScheduleValues[number];

export const discussionFlagValues = ['Flag as safety risk', 'Flag for discussion'] as const;
export type DiscussionFlag = typeof discussionFlagValues[number];

export const cancerTreatmentRegimenValues = [
    'Surgery',
    'Chemotherapy',
    'Radiation',
    'Stem Cell Transplant',
    'Immunotherapy',
    'CAR-T',
    'Endocrine',
    'Surveillance',
] as const;
export type CancerTreatmentRegimen = typeof cancerTreatmentRegimenValues[number] | OtherSpecify;

export const referralValues = [
    'Psychiatry',
    'Psychology',
    'Patient Navigation',
    'Integrative Medicine',
    'Spiritual Care',
    'Palliative Care',
] as const;
export type Referral = typeof referralValues[number];

export const referralStatusValues = ['Not Referred', 'Pending', 'Active', 'Completed'] as const;
export type ReferralStatus = typeof referralStatusValues[number];

export const sessionTypeValues = ['In person', 'Telehealth', 'Phone', 'Group', 'Home'] as const;
export type SessionType = typeof sessionTypeValues[number] | OtherSpecify;

export const treatmentPlanValues = [
    'Maintain current treatment',
    'Adjust treatment plan',
    'Monitor only',
    'No further follow-up',
    'Refer to community',
] as const;
export type TreatmentPlan = typeof treatmentPlanValues[number];

export const treatmentChangeValues = ['None', 'Medication', 'Counseling'] as const;
export type TreatmentChange = typeof treatmentChangeValues[number] | OtherSpecify;

export const behavioralActivationChecklistValues = [
    'Review of the BA model',
    'Values and goals assessment',
    'Activity scheduling',
    'Mood and activity monitoring',
    'Relaxation',
    'Positive reinforcement',
    'Managing avoidance behaviors',
    'Problem-solving',
] as const;
export type BehavioralActivationChecklistItem = typeof behavioralActivationChecklistValues[number];

export const behavioralStrategyChecklistValues = [
    'Behavioral Activation',
    'Motivational Interviewing',
    'Problem Solving Therapy',
    'Cognitive Therapy',
    'Mindfulness Strategies',
    'Supportive Therapy',
] as const;
export type BehavioralStrategyChecklistItem = typeof behavioralStrategyChecklistValues[number] | OtherSpecify;

export const assessmentFrequencyValues = ['Once a week', 'Every 2 weeks', 'Every 4 weeks'] as const;
export type AssessmentFrequency = typeof assessmentFrequencyValues[number];

export const phq9ItemValues = [
    'Interest',
    'Mood',
    'Sleep',
    'Energy',
    'Appetite',
    'Guilt',
    'Concentrating',
    'Motor',
    'Suicide',
] as const;
export type PHQ9Item = typeof phq9ItemValues[number];

export const gad7ItemValues = [
    'Anxious',
    'Constant worrying',
    'Worrying too much',
    'Trouble relaxing',
    'Restless',
    'Irritable',
    'Afraid',
] as const;
export type GAD7Item = typeof gad7ItemValues[number];

export type CaseReviewEntryType = 'Case Review';
export type SessionEntryType = 'Session';
export type EntryType = SessionEntryType | CaseReviewEntryType;

export const daysOfWeekValues = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'] as const;

export type DayOfWeek = typeof daysOfWeekValues[number];
export type DayOfWeekFlags = { [item in DayOfWeek]: boolean };

export const activitySuccessTypeValues = ['Yes', 'SomethingElse', 'No'];
export type ActivitySuccessType = typeof activitySuccessTypeValues[number];

export type BAChecklistFlags = { [item in BehavioralActivationChecklistItem]: boolean };
export type BehavioralStrategyChecklistFlags = { [item in BehavioralStrategyChecklistItem]: boolean };
export type ReferralStatusFlags = { [item in Referral]: ReferralStatus };

export type CancerTreatmentRegimenFlags = { [item in CancerTreatmentRegimen]: boolean };
export type DiscussionFlags = { [item in DiscussionFlag]: boolean };

export const dueTypeValues = ['Exact', 'ChunkOfDay', 'Day', 'Week'];
export type DueType = typeof dueTypeValues[number];

export const providerRoleValues = ['studyStaff', 'psychiatrist', 'socialWorker'];
export type ProviderRole = typeof providerRoleValues[number];

export const siteValues = ['SCCA – SLU', 'SCCA – UW-NW', 'SCCA – PEN', 'MultiCare – Tacoma', 'MultiCare – Gig Harbor'];
export type Site = typeof siteValues[number];
